File: core/weapons/pulse-canon.js
import { assets } from "../assets/main.js";
import { worldSize } from "../utils/constants.js";
import { wrap } from "../world/utils.js";
import Projectile from "./projectile.js";

export default class PulseCanon extends Projectile {
    constructor({ x, y, angle, ship, color, speed = 10 }) {
        super({
            name: "Pulse Canon",
            speed: speed,
            acceleration: 50000,
            x: x,
            y: y,
            width: 8,
            height: 15,
            angle: angle,
            damage: 40,
            range: 10000,
            fireRate: 0.5,
            energyCost: 50,
            ship,
            color,
            img: assets?.images?.projectile
        });
    }
}

File: void_front/game/core/weapons/pulse_canon.py
from __future__ import annotations

from game.core.weapons.projectile import Projectile


class PulseCanon(Projectile):
    def __init__(self, options, force=False):
        defaults = {
            "name": "Pulse Canon",
            "speed": options.get("speed", 10),
            "acceleration": 50000,
            "x": options.get("x"),
            "y": options.get("y"),
            "width": 8,
            "height": 15,
            "angle": options.get("angle"),
            "damage": 40,
            "range": 10000,
            "fireRate": 0.5,
            "energyCost": 50,
            "ship": options.get("ship"),
            "color": options.get("color"),
            "img": options.get("img"),
        }
        super().__init__(defaults, force)
